src/Xenial.Tasty/Execution/TestRuntime/ReportSummaryMiddleware.cs

```diff
@@ -17,11 +16,25 @@ public static TestExecutor UseSummaryReporters(this TestExecutor executor)
                 }
                 finally
                 {
-                    var cases = context.Scope.Descendants().OfType<TestCase>().ToList();
+                    async Task<IEnumerable<TestSummary>> Summaries()
```

Comment:
Should we use an `IAsyncEnumerable` here?

Reply:
Possible, but I couldn't make it work properly

Reply:
```cs
public static class ReportSummaryMiddleware
    {
        public static TestExecutor UseSummaryReporters(this TestExecutor executor)
            => executor.UseRuntime(async (context, next) =>
            {
                try
                {
                    await next();
                }
                finally
                {
                    async IAsyncEnumerable<TestSummary> Summaries()
                    {
                        foreach (var provider in context.Scope.SummaryProviders)
                        {
                            yield return await provider.Invoke();
                        }
                    }

                    await Task.WhenAll(context.Scope.SummaryReporters
                        .Select(async r =>
                        {
                            await foreach (var summary in Summaries())
                            {
                                await r.Invoke(summary);
                            }
                        }).ToArray());
                }
            });
    }
```